app/controllers/application_controller.rb

```diff
@@ -2152,24 +2152,38 @@ def build_saved_audit(rec, eh)
     }
   end
 
-  def prov_redirect(typ=nil)
-    assert_privileges(params[:pressed])
-    if typ  # we need to do this check before doing anything to prevent
-            # history being updated
-      vm_ids = find_checked_items.map(&:to_i).uniq
-      if !typ.eql?("clone") && VmOrTemplate.includes_template?(vm_ids)
+  def task_supported?(typ)
```

Comment:
Furthermore, `render_flash_not_applicable_to_model` assumes MiqTemplate as the model in `:model => ui_lookup(:table => "miq_template")`